crates/noirc_evaluator/src/ssa_refactor/opt/defunctionalize.rs

```diff
@@ -58,26 +42,35 @@ struct ApplyFunction {
     dispatches_to_multiple_functions: bool,
 }
 
+fn signature_from_call_params(
+    caller: &Function,
+    params: &[ValueId],
+    results: &[ValueId],
+) -> Signature {
+    let params = vecmap(params, |param| caller.dfg.type_of_value(*param));
+    let returns = vecmap(results, |result| caller.dfg.type_of_value(*result));
+    Signature { params, returns }
```

Comment:
Looks like mostly a copy of the From impl. Either we only need one of these or the From impl can reuse this logic.

Reply:
It was just used in two places and not gaining much, so I just removed it